src/mongocxx/index_view.hpp

```diff
@@ -38,14 +30,11 @@ MONGOCXX_INLINE_NAMESPACE_BEGIN
 
 class MONGOCXX_API index_view {
    public:
-    index_view() = delete;
-
-    index_view(const index_view&) = delete;
+    index_view(index_view&&) noexcept;
+    index_view& operator=(index_view&&) noexcept;
```

Comment:
Give index_view both move ctor and move assign, and make them properly noexcept.